app/components/steps/why.js

```diff
@@ -0,0 +1,254 @@
+//https://github.com/EnCiv/civil-pursuit/issues/103
+//https://github.com/EnCiv/civil-pursuit/issues/214
+
+import React, { useEffect, useContext, useRef, useState } from 'react'
+import DeliberationContext from '../deliberation-context'
+import WhyInput from '../why-input'
+import { H, Level } from 'react-accessible-headings'
+import cx from 'classnames'
+import { createUseStyles } from 'react-jss'
+import { isEqual, cloneDeep } from 'lodash'
+
+export default function WhyStep(props) {
+  const { data, upsert } = useContext(DeliberationContext)
+  const { category, intro, ...otherProps } = props
+  const derivedPointWhyList = derivePointWhyListByCategory(data, category)
+
+  useEffect(() => {
+    if (!data?.myWhyByParentId && data?.reducedPointList?.length > 0) {
+      const ids = data.reducedPointList.map(point => point._id)
+      window.socket.emit('getUserWhys', ids, results => {
+        const myWhyByParentId = results.reduce((acc, point) => {
+          acc[point.parentId] = point
+          return acc
+        }, {})
+        upsert({ myWhyByParentId })
+      })
+    }
+  }, [data, upsert])
+
+  function handleOnDone({ valid, value, delta }) {
+    if (!delta.category) {
+      delta.category = category
+    }
+
+    const updatedMyWhyByParentId = {
```

Comment:
Generally they teach to do it this way, to make a copy of the previous state and add in the change. But for this situation, we are carefully only changing  what's changed and it's parent refs so the React will only rerender what's changed. 

So similar to rerank.js  this should just upsert({myWhyByParentId: {delta.parentId: delta}}).  upsert will take care of changing what else needs to change.

Also, you really really needs to `git fetch` and `git merge origin/master` to get the new version of upsert and other changes.